prow/git/git.go

```diff
@@ -1,639 +0,0 @@
-	c.rlm.Unlock()
-	m.Lock()
-}
-
-func (c *Client) unlockRepo(repo string) {
-	c.rlm.Lock()
-	defer c.rlm.Unlock()
-	c.repoLocks[repo].Unlock()
-}
-
-func remoteFromBase(base, user, pass, host, org, repo string) string {
-	baseWithAuth := base
-	if user != "" && pass != "" {
-		baseWithAuth = fmt.Sprintf("https://%s:%s@%s", user, pass, host)
-	}
-	return fmt.Sprintf("%s/%s/%s", baseWithAuth, org, repo)
-}
-
-// Clone clones a repository. Pass the full repository name, such as
-// "kubernetes/test-infra" as the repo.
-// This function may take a long time if it is the first time cloning the repo.
-// In that case, it must do a full git mirror clone. For large repos, this can
-// take a while. Once that is done, it will do a git fetch instead of a clone,
-// which will usually take at most a few seconds.
-func (c *Client) Clone(organization, repository string) (*Repo, error) {
-	orgRepo := organization + "/" + repository
-	c.lockRepo(orgRepo)
-	defer c.unlockRepo(orgRepo)
-
-	user, pass, err := c.getCredentials(organization)
-	if err != nil {
-		return nil, fmt.Errorf("failed to get token: %w", err)
-	}
-	cache := filepath.Join(c.dir, orgRepo) + ".git"
-	remote := remoteFromBase(c.base, user, pass, c.host, organization, repository)
-	if _, err := os.Stat(cache); os.IsNotExist(err) {
-		// Cache miss, clone it now.
-		c.logger.WithField("repo", orgRepo).Info("Cloning for the first time.")
-		if err := os.MkdirAll(filepath.Dir(cache), os.ModePerm); err != nil && !os.IsExist(err) {
-			return nil, err
-		}
-		if b, err := retryCmd(c.logger, "", c.git, "clone", "--mirror", remote, cache); err != nil {
-			return nil, fmt.Errorf("git cache clone error: %v. output: %s", err, string(b))
-		}
-	} else if err != nil {
-		return nil, err
-	} else {
-		// Cache hit. Do a git fetch to keep updated.
-		// Update remote url, if we use apps auth the token changes every hour
-		if b, err := retryCmd(c.logger, cache, c.git, "remote", "set-url", "origin", remote); err != nil {
-			return nil, fmt.Errorf("updating remote url failed: %w. output: %s", err, string(b))
-		}
-		c.logger.WithField("repo", orgRepo).Info("Fetching.")
-		if b, err := retryCmd(c.logger, cache, c.git, "fetch", "--prune"); err != nil {
-			return nil, fmt.Errorf("git fetch error: %v. output: %s", err, string(b))
-		}
-	}
-	t, err := os.MkdirTemp("", "git")
-	if err != nil {
-		return nil, err
-	}
-	if b, err := exec.Command(c.git, "clone", cache, t).CombinedOutput(); err != nil {
-		return nil, fmt.Errorf("git repo clone error: %v. output: %s", err, string(b))
-	}
-	// Updating remote url to true remote like `git@github.com:kubernetes/test-infra.git`,
-	// instead of something like `/tmp/12345/test-infra`, so that `git fetch` in this clone makes more sense.
-	cmd := exec.Command(c.git, "remote", "set-url", "origin", remote)
-	cmd.Dir = t
-	if b, err := cmd.CombinedOutput(); err != nil {
-		return nil, fmt.Errorf("updating remote url failed: %w. output: %s", err, string(b))
-	}
```

Comment:
Note: We don't need to do this anymore, the V2 Fetch functions know how to update the primary clone before fetching (if needed).